src/Uno.UI.RemoteControl.VS/SimpleServiceProvider.cs

```diff
@@ -0,0 +1,38 @@
+using System;
+using System.Collections.Immutable;
+using IAsyncDisposable = Microsoft.VisualStudio.Threading.IAsyncDisposable;
+
+namespace Uno.UI.RemoteControl.VS;
+
+internal class SimpleServiceProvider : IServiceProvider, IDisposable
+{
+	private ImmutableDictionary<Type, object> _services = ImmutableDictionary<Type, object>.Empty;
+
+	public void Register(Type contract, object instance)
```

Comment:
Looks like a potential NRE here. What happends if contract\instance is null here?